### PR TITLE
Batch ingest as rake task

### PR DESCRIPTION
**Pull request by aploshay**
Code to run batch ingest as a rake task with:
rake batch_ingest

Other differences:
- Doesn't use FactoryGirl (although that code is still present)
- Improves performance with unchecked page saves
- Adds import of pagedXML file

Pending:
- There's some amount of package validation in there, that should probably all be stripped out and just dropped or put elsewhere in a validation module
- There's no cleanup (deletion) of the ingested packages yet

### CODE REVIEW

**Comment by randalldfloyd** [comment]
I can't quite tell from following the code if a final save or update_index is triggered on the paged object after pages are added. This would be required to create the solr field that is used to send json page data to the bookreader. It does unfortunately cause at least one paged.pages to occur but it's unavoidable to get all the child pages into that field.

**Comment by aploshay** [comment]
Oh, that's totally missing; I'll add an update_index call and include the additional commit in the revised pull request.

**Comment by aploshay** [comment]
I pushed the additional commit, and it tested okay manually, so I think we're good now.

**Comment by mwoodiupui** [comment]
I'm not convinced that we should delete an ingestion package as part of ingestion.  If not doing so causes a problem, then I think that we should (as a separate issue) discover why this is so.

File: lib/tasks/batch_import.rb [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  begin
+    paged_yaml["descMetadata"].each_pair do |key, value|
+      paged_attributes[key.to_sym] = value
+    end
+  rescue
+    puts "ABORTING PAGED CREATION: invalid structure in descMetadata"
+  end
+  if paged_attributes.any?
+    begin
+      paged = Paged.new(**paged_attributes)
```
Comment (mwoodiupui) [inline lib/tasks/batch_import.rb]: Don't really need the double-splat here:  there's nothing for paged_attributes to be merged with.

**Comment by randalldfloyd** [comment]
Everything looks and works great with one minor exception. The manifest format and the original factory driven ingest drifted just a little in the last couple of sprints. We were using the package ingest for now to seed the structured fields to drive hierarchical facets (paged_struct/page_struct in both the paged and page objects), which you can see in https://github.com/IU-Libraries-Joint-Development/PagedMedia/blob/develop/spec/fixtures/ingest/pmp/package1/manifest.yml .  You can simply pull them out from within the descMetadata section in the same way that you are extracting other metadata fields.

**Comment by randalldfloyd** [comment]
I see what's happening now. The code doesn't literally spell out attributes to assign but reads them all in from the manifest and expects them to match model attributes. The paged_struct/page_struct attributes were also added to the models later so they aren't in this pull request. Theoretically it will work once this is merged with current develop.

**Comment by aploshay** [comment]
Okay, I'm in the process of updating the manifest.yml and merging the current develop into this branch to test which things break.

**Comment by randalldfloyd** [comment]
+1 Works great now. My 70 page score only took about 3 minutes on my slow virtual machine.

**Pull request merged**